user/models.py

```diff
@@ -44,5 +63,12 @@ def __str__(self):
         return str(self.username or 'Unknown')
 
     def save(self, *args, **kwargs):
+        try:
+            User.objects.get(username=self.username)
```

Comment:
Notice that we don't really mind if the username is unique, but rather the slug (that's the one that crashes).